projects/scannet_offline_eval/build_sparse_voxel_map.py

```diff
@@ -0,0 +1,291 @@
+            obs = Observations(
+                rgb=obs.cpu().numpy(), gps=None, compass=None, depth=None
+            )
+            res = self.semantic_sensor.predict(obs).task_observations
+            instance_image = torch.from_numpy(res["instance_map"]).int().to(device)
+            instance_classes = (
+                torch.from_numpy(res["instance_classes"]).int().to(device)
+            )
+            instance_scores = torch.from_numpy(res["instance_scores"]).int().to(device)
+            # semantic_frame = torch.from_numpy(res['semantic_frame']
+        else:
+            instance_image = obs.instance
+            instance_classes = obs.task_observations["instance_classes"]
+            instance_scores = obs.task_observations["instance_scores"]
+
+        self.voxel_map.add(
+            rgb=obs.rgb.float() / 255.0,
+            depth=obs.depth.squeeze(-1),
+            feats=obs.task_observations["features"],
+            camera_K=obs.camera_K,
+            camera_pose=obs.camera_pose,  # scene_obs['axis_align_mats'][i] @ scene_obs['poses'][i],
+            instance_image=instance_image,
+            instance_classes=instance_classes,
+            instance_scores=instance_scores,
+        )
+
+        if visualize_map:
+            # Now draw 2d
+            self.voxel_map.get_2d_map(debug=True)
+
+    def get_2d_map(self):
+        """Get 2d obstacle map for low level motion planning and frontier-based exploration"""
+        return self.voxel_map.get_2d_map()
+
+    def show(self) -> Tuple[np.ndarray, np.ndarray]:
+        """Display the aggregated point cloud."""
+        return self.voxel_map.show(
+            instances=True,
+            height=1000,
+            boxes_plot_together=False,
+            boxes_name_int_to_display_name_dict=dict(
+                enumerate(self.metadata.thing_classes)
+            ),
+            backend="pytorch3d",
+        )
+
+
+# TODO: use this code to create semantic sensor
+# def create_semantic_sensor(
+#     device_id: int = 0,
+#     verbose: bool = True,
+#     **kwargs,
+# ):
+#     """Create segmentation sensor and load config. Returns config from file, as well as a OvmmPerception object that can be used to label scenes."""
+#     print("- Loading configuration")
+#     config = load_config(visualize=False, **kwargs)
+
+#     print("- Create and load vocabulary and perception model")
+#     semantic_sensor = OvmmPerception(config, device_id, verbose, module="detic")
+#     obj_name_to_id, rec_name_to_id = read_category_map_file(
+#         config.ENVIRONMENT.category_map_file
+#     )
+#     vocab = build_vocab_from_category_map(obj_name_to_id, rec_name_to_id)
+#     semantic_sensor.update_vocabulary_list(vocab, 0)
+#     semantic_sensor.set_vocabulary(0)
+#     return config, semantic_sensor
+
+
+if __name__ == "__main__":
+    from eval.utils import eval_bboxes_and_print
+
+    parser = argparse.ArgumentParser(
```

Comment:
I've been using click, i dont know if we care to make it consistent? this is fine

Reply:
Yeah I think we should use click or hydra for this. Hydra probably makes it easier to maintain/test

Reply:
hydra is probably better i agree

Reply:
we don't need to do this now though